### PR TITLE
feature: better error messages on provider api keys and commits actions

### PR DESCRIPTION
**Pull request by geclos**

### CODE REVIEW

File: packages/core/src/lib/errors.ts [hunk 1]
```diff
@@ -50,3 +50,8 @@ export class UnauthorizedError extends LatitudeError {
   public statusCode = 401
   public name = 'UnauthorizedError'
 }
+
+export const databaseErrorCodes = {
+  foreignKeyViolation: '23503',
+  uniqueViolation: '23505',
+}
```
Comment (andresgutgon) [inline packages/core/src/lib/errors.ts:57]: This should be here https://github.com/latitude-dev/latitude-llm/blob/main/packages/core/src/lib/Transaction.ts#L21

File: packages/core/src/services/providerApiKeys/destroy.ts [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
+        .returning()
+      const deleted = result[0]
 
-    return Result.ok(deleted)
+      return Result.ok(deleted)
+    } catch (error) {
+      if (
+        error instanceof DatabaseError &&
+        error.code === databaseErrorCodes.foreignKeyViolation
+      ) {
```
Comment (andresgutgon) [inline packages/core/src/services/providerApiKeys/destroy.ts:31]: This is not the right way of handle this. We don't have to wait to arrive to the DB to fail. This has to be validated in the action with a zod async validation. 

Example:
https://github.com/latitude-dev/latitude-llm/blob/main/apps/web/src/actions/user/setupAction.ts#L22-L25

Comment (andresgutgon) [reply packages/core/src/services/providerApiKeys/destroy.ts:31]: This is like if do a catch in a Rails transaction. We we would do that having models and validations?

Comment (andresgutgon) [reply packages/core/src/services/providerApiKeys/destroy.ts:31]: Please change it to don't set a bad precent

Comment (geclos) [reply packages/core/src/services/providerApiKeys/destroy.ts:31]: i disagree why add an extra db roundtrip to all requests for a validation when you can just let the db validation fail and return a nicely formatted error 🤷🏼

File: packages/core/src/lib/errors.ts (hunk 1 above)
Comment (geclos) [reply packages/core/src/lib/errors.ts:57]: will move even if this is just ocd

File: packages/core/src/services/providerApiKeys/destroy.ts (hunk 2 above)
Comment (geclos) [reply packages/core/src/services/providerApiKeys/destroy.ts:31]: i absolutely do not think this is a bad precedent

Comment (andresgutgon) [reply packages/core/src/services/providerApiKeys/destroy.ts:31]: But this is exactly how Rails works. Is a validation check againts Db. Why matters the roundtrip in a POST operation? I don't think is the bottle neck.

Also services should receive valid data the same way they receive valid scopes.

Having to deal with try/catch feels nasty. Having zod which provides a nice layer between client and server

File: packages/core/src/lib/errors.ts (hunk 1 above)
Comment (andresgutgon) [reply packages/core/src/lib/errors.ts:57]: How is this ocd? Is the same unique violation error in two different places. For that matter, put it in the destroy service as a local constant

File: packages/core/src/services/providerApiKeys/destroy.ts (hunk 2 above)
Comment (andresgutgon) [reply packages/core/src/services/providerApiKeys/destroy.ts:31]: Any way. I said my part feel free to ignore

**Pull request merged**